tests/end2end/features/completion.feature

```diff
@@ -0,0 +1,80 @@
+Feature: Using completion
+
+    Scenario: Using command completion
+        When I run :set-cmd-text :
+        Then the completion model should be CommandCompletionModel
+
+    Scenario: Using help completion
+        When I run :set-cmd-text -s :help
+        Then the completion model should be HelpCompletionModel
+
+    Scenario: Using quickmark completion
+        When I run :set-cmd-text -s :quickmark-load
+        Then the completion model should be QuickmarkCompletionModel
+
+    Scenario: Using bookmark completion
+        When I run :set-cmd-text -s :bookmark-load
+        Then the completion model should be BookmarkCompletionModel
+
+    Scenario: Using bind completion
+        When I run :set-cmd-text -s :bind X
+        Then the completion model should be BindCompletionModel
+
+    Scenario: Using session completion
+        Given I open data/hello.txt
+        And I run :session-save hello
+        When I run :set-cmd-text -s :session-load
+        And I run :completion-item-focus next
+        And I run :completion-item-focus next
+        And I run :session-delete hello
+        And I run :command-accept
+        Then the error "Session hello not found!" should be shown
+
+    Scenario: Using option completion
+        When I run :set-cmd-text -s :set colors
+        Then the completion model should be SettingOptionCompletionModel
+
+    Scenario: Using value completion
+        When I run :set-cmd-text -s :set colors statusbar.bg
+        Then the completion model should be SettingValueCompletionModel
+
+    Scenario: Using value completion multiple times
+        When I run :set-cmd-text -s :set --cycle colors statusbar.bg black
+        Then the completion model should be SettingValueCompletionModel
+
+    Scenario: Updating the completion in realtime
+        Given I have a fresh instance
+        And I set completion -> quick-complete to false
+        When I open data/hello.txt
+        And I run :set-cmd-text -s :buffer
+        And I run :completion-item-focus next
+        And I open data/hello2.txt in a new background tab
+        And I run :completion-item-focus next
+        And I open data/hello3.txt in a new background tab
+        And I run :completion-item-focus next
+        And I run :command-accept
+        Then the following tabs should be open:
+            - data/hello.txt
+            - data/hello2.txt
+            - data/hello3.txt (active)
+
+    Scenario: Updating the value completion in realtime
+        Given I set colors -> statusbar.bg to green
+        When I run :set-cmd-text -s :set colors statusbar.bg
+        And I set colors -> statusbar.bg to yellow
```

Comment:
As previous, the exact nature of the test is less important than the fact that it's testing the underlying update mechanism. Either way, this is a deliberate behavior (otherwise a config change would not trigger the completion to update), and therefore a test vector deserves to exist for it.

You might as well question the validity of having the :set completion update itself when the setting changes externally, but that would be a criticism in a different part of the code.